registered user can log in and out

damwhit opened the pull request:
a registered user has the ability to log in and out of an existing account

damwhit committed:
registered user can log in and out

a registered user has the ability to log in and out of an existing account

erinnachen commented on spec/features/user/user_can_login_and_out_of_existing_accounts_spec.rb:
```diff
@@ -0,0 +1,38 @@
+    end
+
+    expect(current_path).to eq("/users/#{user.id}/dashboard")
+
```
could we make this route  '/dashboard' ?

erinnachen commented on spec/features/user/user_can_login_and_out_of_existing_accounts_spec.rb:
```diff
@@ -0,0 +1,38 @@
+    expect(page).to have_link("order history")
+
+    click_on "logout"
+
```
Is there a flash message that should be displayed when one logs out?

damwhit committed:
user can log in and out.

added validation for flash and to see the user's name within the sidebar nav and sad path for both user log in and registrations are now tested

damwhit commented:
asserted logout flash in test and changed user dashboard path to be '/dashboard'

damwhit commented:
added sad paths

cheljoh commented:
Looking good Davey, thanks for adding sad paths!
